src/Support/ResolveInfoFieldsAndArguments.php

```diff
@@ -130,11 +130,11 @@ private function foldSelectionSet(SelectionSetNode $selectionSet, int $descend):
                 $spreadName = $selectionNode->name->value;
                 if (isset($this->info->fragments[$spreadName])) {
                     $fragment = $this->info->fragments[$spreadName];
-                    $fields = array_merge_recursive($this->foldSelectionSet($fragment->selectionSet, $descend),
+                    $fields = (array) array_replace_recursive($this->foldSelectionSet($fragment->selectionSet, $descend),
```

Comment:
What do we need the casts?

Seems to me it works without them?

Reply:
At some point, phpstan failed because of not having cast. `array_replace_recursive` can return array or null (in case of error, but I couldn't find more details about when that happens and why), but `array_merge_recursive` always returns array. So at the end, we need casting to array now.